sips/sip-011/sip-011-catastrophic-failure-recovery.md

```diff
@@ -0,0 +1,464 @@
+miners through some other channel.
+* Declaring a "flag day" for a subsequent hard fork, by declaring that the
+  current node software will reach end-of-life status by a certain burnchain
+block height.  This may be appropriate if the bugfix for the catastrophic
+failure is temporary (e.g. if the design of concensus-critical code-paths in the
+blockchain are at fault, and preclude a permanent fix).
+
+There are multiple different ways to execute a soft fork.  This document only
+discusses procedures for a _miner-actived_ soft fork whose goal is to address a
+liveness or safety failure.  A subsequent SIP may codify other types of soft
+forks and their applications in different contexts.  Due to the comparatively
+low barrier-to-entry for mining on the Stacks chain, miner-activated soft forks
+are expected to offer an acceptable measurement of the ecosystem's sentiments on
+a pending soft fork.
+
+A miner-activated soft fork procedure for healing a catastrophic failure works
+as follows:
+
+1. The Foundation and at least one other significant ecosystem entity will approve and
+sign a Stacks node software release that implements the soft fork rules.  This
+is the first step in this process because this new node software must address
+whatever catastrophic bug that led to the need to follow this procedure.  The
+remainder of this procedure is focused on codifying the new rules via the SIP
+process in order to require them in future node software releases.  The act of
+releasing this new node software will follow the critical bugfix point-release
+procedure, but with an additional warning from the announcement email that this
+new release is the beginning of a soft-fork roll-out.
+2. Once the catastrophic bug has been addressed, and a reliable majority of
+blocks on the canonical fork are produced by miners running the new software, a
+SIP describing the soft fork will be written and published describing the new
+rules and why they are necessary.  This SIP must then be advanced in the SIP
+process to "activation-in-progress" status by the relevant advisory boards and
+technical steering committee.  Among other things, this SIP's activation
+criteria must require the following:
+
+   * At least 80% of all miners must _explicitly and continuously vote to
+     accept_ the soft fork rules for at least _two consecutive_ whole PoX reawrd
+cycles. Additional, stricter criteria are permitted.
+   * An _activation window_, defined by a _start activation height_ and an _end
+     activation height_, represented as burnchain block heights, must be
+stipulated in the activation criteria.  The aforementioned miner vote must take
+place and run to completion in-between these two block heights.  The block
+heights _must_ correspond to the beginnings of PoX reward cycles, and must be
+between 14 reward cycles (1 maximum Stacking period plus two reward cycles) and
+25 reward cycles (about 1 year) apart.
```

Comment:
Is this too long? I'm imagining a situation where say two soft-forks are needed, 6-months apart. How do the start/end activation heights work in this case (they would overlap)?

Reply:
Multiple soft forks can happen concurrently if they don't conflict or aren't dependent on one another.  This sentence is only a statement about the possible values the activation window block heights can take.

Reply:
Addressed in 3832ae2